WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ButtonStyles.swift

```diff
@@ -0,0 +1,103 @@
+import SwiftUI
+
+struct PrimaryButtonStyle: ButtonStyle {
+    func makeBody(configuration: Configuration) -> some View {
+        PrimaryButton(configuration: configuration)
+    }
+}
+
+struct SecondaryButtonStyle: ButtonStyle {
+    func makeBody(configuration: Configuration) -> some View {
+        SecondaryButton(configuration: configuration)
+    }
+}
+
+private struct BaseButton: View {
+    let configuration: ButtonStyleConfiguration
+
+    var body: some View {
+        configuration.label
+            .frame(maxWidth: .infinity)
+            .font(.headline)
+            .padding(Style.defaultEdgeInsets)
+    }
+}
+
+private struct PrimaryButton: View {
+    @Environment(\.isEnabled) var isEnabled
+
+    let configuration: ButtonStyleConfiguration
+
+    var body: some View {
+        BaseButton(configuration: configuration)
+            .foregroundColor(Color(isEnabled ? .primaryButtonTitle : .buttonDisabledTitle))
+            .background(
+                RoundedRectangle(cornerRadius: Style.defaultCornerRadius)
+                    .fill(Color(isEnabled
+                                        ? configuration.isPressed ? .primaryButtonDownBackground : .primaryButtonBackground
+                                        : .buttonDisabledBackground))
```

Comment:
For these color configurations do you think it's worth it to have something like
```swift
private var enabledColor: UIColor {
    configuration.isPressed ? .primaryButtonDownBackground : .primaryButtonBackground
}

private var disabledColor: UIColor {
    .buttonDisabledBackground
}
```

For then being able to configure it like:
```swift
.background(
    RoundedRectangle(cornerRadius: Style.defaultCornerRadius)
        .fill(Color(isEnabled ? enabledColor : disabledColor))
)
```

I mention this because I found really hard to know what color was being set when I was reading the code.

Reply:
That makes sense, the nested ternaries are not very readable 😞

Reply:
I've extracted those to computed properties, and used if/else instead of ternaries in d0e750b5c35ecc37d5318c89c6a8e60675807d78, which I think reads a lot clearer now